opennlp-tools-models/src/main/java/opennlp/tools/models/ClasspathModelFinder.java

```
@@ -0,0 +1,121 @@
+   * @return A Set of {@link ClassPathModelEntry ClassPathModelEntries}. It might be empty.
+   */
+  public Set<ClassPathModelEntry> findModels(boolean reloadCache) {
+
+    if (this.models == null || reloadCache) {
+      try (ScanResult sr = new ClassGraph().acceptJars(jarModelPrefix).disableDirScanning().scan()) {
+
+        final List<URI> classpathModels = getResourcesMatchingWildcard(sr, "*.bin");
+        final List<URI> classPathProperties = getResourcesMatchingWildcard(sr, "model.properties");
+
+        this.models = new HashSet<>();
+
+        for (URI model : classpathModels) {
+          URI m = null;
+          for (URI prop : classPathProperties) {
+            if (jarPathsMatch(model, prop)) {
+              m = prop;
+              break;
+            }
+          }
+          this.models.add(new ClassPathModelEntry(model, Optional.ofNullable(m)));
+
+        }
+      }
+    }
+    return this.models;
+  }
+
+  private List<URI> getResourcesMatchingWildcard(final ScanResult sr, final String resourceWildcard) {
+    try (final ResourceList resources = sr.getResourcesMatchingWildcard(resourceWildcard)) {
+      return resources.getURIs();
+    }
+  }
+
+  private boolean jarPathsMatch(URI uri1, URI uri2) {
+    final String[] parts1 = parseJarURI(uri1);
+    final String[] parts2 = parseJarURI(uri2);
+
+    if (parts1 == null || parts2 == null) {
+      return false;
+    }
+
+    return parts1[0].equals(parts2[0]);
+  }
+
+  private String[] parseJarURI(URI uri) {
+    try {
+      if ("jar".equals(uri.getScheme())) {
+        final String ssp = uri.getSchemeSpecificPart();
+        final int separatorIndex = ssp.indexOf("!/");
+        if (separatorIndex > 0) {
+          final String jarFileUri = ssp.substring(0, separatorIndex);
+          final String entryPath = ssp.substring(separatorIndex + 2);
+          return new String[] {jarFileUri, entryPath};
+        }
+      }
+    } catch (Exception ignored) {
```

Pls leave a comment, why it safe to ignore all exceptions here.